includes/User.php

```diff
@@ -2082,41 +2062,42 @@ public static function clearUserCache( $userId ) {
 	 * wipes it, so the account cannot be logged in until
 	 * a new password is set, for instance via e-mail.
 	 *
-	 * @param $str String New password to set
-	 * @throws PasswordError on failure
+	 * @param string $password New password to set
+	 * @param bool   $forceLogout
 	 *
-	 * @return bool
+	 * @return bool on failure
+	 *
+	 * @throws Exception
+	 * @throws PasswordError on failure
 	 */
-	public function setPassword( $str, $forceLogout = true ) {
-		global $wgAuth;
-
-		if( $str !== null ) {
-			if( !$wgAuth->allowPasswordChange() ) {
-				throw new PasswordError( wfMsg( 'password-change-forbidden' ) );
-			}
+	public function setPassword( $password, $forceLogout = true ) {
+		$heliosClient = self::heliosClient();
+		$heliosPasswordChange = null;
 
-			if( !$this->isValidPassword( $str ) ) {
```

Comment:
Do we check this on the Helios side when setting a password now? What if Helios fails and we set this in MediaWiki as is still being done as well, will this have been checked?